docs/design/configuration.md

```diff
@@ -0,0 +1,141 @@
+# Kubernetes Component Configuration
+
+## Abstract
+
+A generic ``Configuration`` object to store command-line flags that is used for
```

Comment:
That said, I don't think this should be only for Kubernetes configuration.

Applications commonly consume configuration via:
* command-line flags
* environment variables
* configuration files

All should be possible, but we should focus on files first.